[BUG] test failed at cmd/probe/internal/binding/etcd/etcd_test.go:125

**Describe the bug**
https://github.com/apecloud/kubeblocks/actions/runs/4089951118/jobs/7053028584

```
--- FAIL: TestGetRole (0.00s)
    etcd_test.go:43: start embedded etcd server error: listen tcp 127.0.0.1:52600: bind: address already in use
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0xebe445]

goroutine 132 [running]:
testing.tRunner.func1.2({0xfb67c0, 0x1a1c340})
	/usr/local/go/src/testing/testing.go:1396 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1399 +0x39f
panic({0xfb67c0, 0x1a1c340})
	/usr/local/go/src/runtime/panic.go:884 +0x212
go.etcd.io/etcd/server/v3/embed.(*Etcd).Close(0x0)
	/runner/_work/kubeblocks/kubeblocks/vendor/go.etcd.io/etcd/server/v3/embed/etcd.go:367 +0x45
github.com/apecloud/kubeblocks/cmd/probe/internal/binding/etcd.(*EmbeddedETCD).Stop(0xc00051a630)
	/runner/_work/kubeblocks/kubeblocks/cmd/probe/internal/binding/etcd/etcd_test.go:125 +0x25
github.com/apecloud/kubeblocks/cmd/probe/internal/binding/etcd.stopEtcdServer(0xfffffffe?)
	/runner/_work/kubeblocks/kubeblocks/cmd/probe/internal/binding/etcd/etcd_test.go:82 +0x1e
panic({0xfb67c0, 0x1a1c340})
	/usr/local/go/src/runtime/panic.go:884 +0x212
github.com/apecloud/kubeblocks/cmd/probe/internal/binding/etcd.(*Etcd).GetRole(0xc000483ba0?, {0x12d22c8?, 0xc0000460e8?}, {0xc000035eb0?, 0x19dff60?})
	/runner/_work/kubeblocks/kubeblocks/cmd/probe/internal/binding/etcd/etcd.go:101 +0x30
github.com/apecloud/kubeblocks/cmd/probe/internal/binding/etcd.TestGetRole(0xc000483ba0)
	/runner/_work/kubeblocks/kubeblocks/cmd/probe/internal/binding/etcd/etcd_test.go:46 +0x118
testing.tRunner(0xc000483ba0, 0x1185b00)
	/usr/local/go/src/testing/testing.go:1446 +0x10b
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:14[93](https://github.com/apecloud/kubeblocks/actions/runs/4089951118/jobs/7053028584#step:5:94) +0x35f
FAIL	github.com/apecloud/kubeblocks/cmd/probe/internal/binding/etcd	0.025s
```

--- a/cmd/probe/internal/binding/etcd/etcd_test.go
+++ b/cmd/probe/internal/binding/etcd/etcd_test.go
@@ -19,8 +19,10 @@ package etcd
 import (
 	"context"
 	"io/ioutil"
+	"math/rand"
 	"net/url"
 	"os"
+	"strconv"
 	"testing"
 	"time"
 
@@ -33,8 +35,10 @@ import (
 
 const (
 	etcdStartTimeout = 30
-	testEndpoint     = "http://localhost:52600"
 )
+
+// randomize the port to avoid conflicting
+var testEndpoint = "http://localhost:" + strconv.Itoa(52600+rand.Intn(1000))
 
 func TestGetRole(t *testing.T) {
 	etcdServer, err := startEtcdServer(testEndpoint)

